src/plugins/Wallet/wallet.ts

```diff
@@ -18,21 +20,16 @@ export function getWalletProvider() {
     return walletAPI
 }
 const memoGetWalletBalance = memoizePromise(
-    (addr: string) => {
-        return getWalletProvider()
-            .queryBalance(addr)
-            .then((x) => onWalletBalanceUpdated(addr, x))
+    async (addr: string) => {
+        const x = await getWalletProvider().queryBalance(addr)
+        return onWalletBalanceUpdated(addr, x)
     },
     (x) => x,
 )
 const memoQueryERC20Token = memoizePromise(
-    (addr: string, erc20Addr: string) => {
-        return getWalletProvider()
-            .queryERC20TokenBalance(addr, erc20Addr)
-            .then((x) => onWalletERC20TokenBalanceUpdated(addr, erc20Addr, x))
-            .catch(() => {
-                // do nothing
-            })
+    async (addr: string, erc20Addr: string) => {
+        const x = await getWalletProvider().queryERC20TokenBalance(addr, erc20Addr)
```

Comment:
Actually the semantics has changed there. This await won't be caught by the noop, but in the original version it is caught by the `.catch` call.